pkg/kn/flags/podspec.go

```diff
@@ -169,6 +173,25 @@ func (p *PodSpecFlags) AddFlags(flagset *pflag.FlagSet) []string {
 func (p *PodSpecFlags) ResolvePodSpec(podSpec *corev1.PodSpec, flags *pflag.FlagSet, allArgs []string) error {
 	var err error
 
+	if flags.Changed("env-file") {
+		envsFromFile, err := util.GetEnvsFromFile(p.EnvFile, "=")
+		if err != nil {
+			return fmt.Errorf("Invalid --env-file: %w", err)
+		}
+		envsFromFileToUpdate, envsFromFileToRemove, err := util.OrderedMapAndRemovalListFromArray(envsFromFile, "=")
+		if err != nil {
+			return fmt.Errorf("Invalid --env: %w", err)
+		}
+		fileEnvsAsArgs := []string{}
+		for _, val := range envsFromFile {
+			fileEnvsAsArgs = append(fileEnvsAsArgs, "-e="+val)
+		}
+		err = UpdateEnvVars(podSpec, fileEnvsAsArgs, envsFromFileToUpdate, envsFromFileToRemove, util.NewOrderedMap(), []string{})
+		if err != nil {
+			return err
+		}
+	}
+
 	if flags.Changed("env") || flags.Changed("env-value-from") {
```

Comment:
i was thinking we could add another || here and add env-file logic here.

Reply:
Yeah, I think I can work it out. Thanks.